R/use_extendr.R

```diff
@@ -108,6 +128,44 @@ use_extendr <- function(path = ".", quiet = getOption("usethis.quiet", FALSE)) {
   return(invisible(TRUE))
 }
 
+#' Checks if provided name is a valid Rust name (identifier)
+#'
+#' @param name \[ character(n) \] Names to test.
+#' @return \[ logical(n) \] `TRUE` if the name is valid, otherwise `FALSE`.
+#' @noRd
+is_valid_rust_name <- function(name) {
+  # We require the name starts with a letter,
+  # ends with a letter or digit,
+  # and contains only alphanumeric ASCII chars, `-` or `_`.
+  stringi::stri_detect_regex(name, "^[A-z][\\w-]*[A-z0-9]$")
```

Comment:
Probably we don't get any non-ASCII characters here, so this is just a nitpicking, but I think `\w` is too wide. On ICU, it means `[\p{Alphabetic}\p{Mark}\p{Decimal_Number}\p{Connector_Punctuation}\u200c\u200d]` ([ref](https://unicode-org.github.io/icu/userguide/strings/regexp.html)) and `\p{Alphabetic}` includes non-ASCII characters. 

``` r
stringi::stri_detect_regex("錆", "\\w")
#> [1] TRUE
```

<sup>Created on 2022-02-17 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>